components/Snackbar/src/MDCSnackbarManager.m

```diff
@@ -286,26 +289,29 @@ - (void)hideSnackbarViewReally:(MDCSnackbarMessageView *)snackbarView
                   [message executeCompletionHandlerWithUserInteraction:userPrompted completion:nil];
                 }];
 
+  __weak MDCSnackbarManagerInternal *weakSelf = self;
   [self.overlayView dismissSnackbarViewAnimated:YES
                                      completion:^{
-                                       self.overlayView.hidden = YES;
-                                       [self deactivateOverlay:self.overlayView];
-
-                                       // If VoiceOver had been enabled and the snackbarView was
-                                       // transient, the Snackbar was just announced (layout was not
-                                       // reported as changed) so there is no need to post a layout
-                                       // change here.
-                                       if (![self isSnackbarTransient:snackbarView]) {
+                                       MDCSnackbarManagerInternal *strongSelf = weakSelf;
+                                       strongSelf.overlayView.hidden = YES;
+                                       [strongSelf deactivateOverlay:strongSelf.overlayView];
+
+                                       // If the snackbarView was transient and
+                                       // accessibilityViewIsModal is NO, the Snackbar was just
+                                       // announced (layout was not reported as changed) so there is
+                                       // no need to post a layout change here.
+                                       if (strongSelf.overlayView.accessibilityViewIsModal ||
+                                           ![strongSelf isSnackbarTransient:snackbarView]) {
                                          UIAccessibilityPostNotification(
                                              UIAccessibilityLayoutChangedNotification, nil);
                                        }
 
-                                       self.currentSnackbar = nil;
+                                       strongSelf.currentSnackbar = nil;
```

Comment:
Are these strong/weak changes required for this PR? Was there previously a leak in the API that we missed?   If not, could they please be moved to a separate PR?  If they are required, please add to the PR summary.